pandas/sparse/array.py

```diff
@@ -266,7 +266,8 @@ def to_dense(self, fill=None):
         # fill the nans
         if fill is None:
             fill = self.fill_value
-        if not np.isnan(fill):
+        # nans can only occur arrays of floating scalars
+        if np.issubdtype(self.dtype, np.inexact) and not np.isnan(fill):
```

Comment:
a better way to do this is just use ``isnull`` which handles all of these things.